### PR TITLE
Move SDK to monorepo

### PR DESCRIPTION
**Pull request by br41nslug**
Moving the JS-SDK into the monorepo.

Replaced `Jest` and `nock` with `vitest` and `msw`.

Fixes ENG-1033

### CODE REVIEW

File: packages/js-sdk/license [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Copyright 2022 Monospace Inc.
```
Comment (rijkvanzanten) [inline packages/js-sdk/license:1]: 2023!

File: packages/js-sdk/package.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	"name": "@directus/sdk",
+	"version": "10.3.3",
+	"description": "The official Directus SDK for use in JavaScript!",
+	"bugs": {
+		"url": "https://github.com/directus/directus/issues"
+	},
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/directus/directus.git",
+		"directory": "packages/js-sdk"
```
Comment (rijkvanzanten) [inline packages/js-sdk/package.json:11]: Lets put this under `packages/sdk`

File: packages/js-sdk/package.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	},
+	"module": "dist/sdk.bundler.js",
+	"unpkg": "dist/sdk.esm.min.js",
+	"types": "dist/index.d.ts",
+	"files": [
+		"dist",
+		"index.mjs"
+	],
+	"scripts": {
+		"build": "tsc --project tsconfig.prod.json",
```
Comment (rijkvanzanten) [inline packages/js-sdk/package.json:33]: Wondering if we should use `unbuild` for this! Maybe another issue 😁

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: packages/js-sdk/package.json (hunk 3 above)
Comment (paescuj) [reply packages/js-sdk/package.json:33]: Yeah, might be a good fit! I think it makes sense to switch in a separate PR 👍

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**